unity-renderer/Assets/Scripts/MainScripts/DCL/Components/Billboard/BillboardsController.cs

```diff
@@ -0,0 +1,92 @@
+using DCL;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BillboardsController : IBillboardsController
+{
+    private readonly List<Billboard> billboards = new List<Billboard>();
+    private Coroutine updateCoroutine;
+
+    private Vector3Variable CameraPosition => CommonScriptableObjects.cameraPosition;
+    private Vector3 lastCamPosition;
+    private bool camUpdated;
+
+
+    public static BillboardsController Create()
+    {
+        BillboardsController controller = new BillboardsController();
+
+        return controller;
+    }
+
+    public void Initialize()
+    {
+        updateCoroutine = CoroutineStarter.Start(UpdateCoroutine());
+    }
+
+    public void BillboardAdded(GameObject billboardContainer)
```

Comment:
In the current implementation you call it from `Billboard` itself, thus you can pass the component directly that makes `TryGetComponent` redundant

Reply:
BillboardsController and Billboard belong to the same assembly, having the interface IBillboardsController knowing about Billboard would create a circular dependency, a solution for this would be to separate BillboardsController and Billboard into different dependencies.

Reply:
The billboard as a Component shouldn't be passed as a reference, instead, I would create an interface for the Billboard itself, you will have even greater decoupling. 

By doing that you can even create a nice test coverage for BillboardsController without spawning game objects